kimera_vio_ros/src/interfaces/stereo_interface.cpp

```diff
@@ -0,0 +1,152 @@
+  // right_sub_.subscribe(node, right_topic, hints.getTransport(), qos.get_rmw_qos_profile());
+  // exact_image_sync_ = std::make_shared<ExactImageSync>(
+  //   ExactImagePolicy(queue_size_), left_sub_, right_sub_);
+
+  // TODO: Assign message filter subscribers to callback_group_stereo_
+  // Pending: https://github.com/ros2/message_filters/issues/45
+  left_image_sub_ = std::make_shared<message_filters::Subscriber<Image>>(
+    node_,
+    left_image_topic,
+    image_qos.get_rmw_qos_profile());
+  right_image_sub_ = std::make_shared<message_filters::Subscriber<Image>>(
+    node_,
+    right_image_topic,
+    image_qos.get_rmw_qos_profile());
+  exact_image_sync_ = std::make_shared<ExactImageSync>(
+    ExactImagePolicy(queue_size_), *left_image_sub_, *right_image_sub_);
+  exact_image_sync_->registerCallback(&StereoInterface::stereo_cb, this);
+  left_image_sub_->unsubscribe();
+  right_image_sub_->unsubscribe();
+}
+
+StereoInterface::~StereoInterface()
+{
+}
+
+void StereoInterface::camera_info_cb(
+  const CameraInfo::ConstSharedPtr & left_msg,
+  const CameraInfo::ConstSharedPtr & right_msg)
+{
+  CHECK_GE(vio_params_->camera_params_.size(), 2u);
+
+  // Initialize CameraParams for pipeline.
+  msgCamInfoToCameraParams(
+    left_msg, &vio_params_->camera_params_.at(0));
+  msgCamInfoToCameraParams(
+    right_msg, &vio_params_->camera_params_.at(1));
+
+  vio_params_->camera_params_.at(0).print();
+  vio_params_->camera_params_.at(1).print();
+
+  // Unregister this callback as it is no longer needed.
+  RCLCPP_INFO(node_->get_logger(),
+    "Unregistering CameraInfo subscribers as data has been received.");
+  left_info_sub_->unsubscribe();
+  right_info_sub_->unsubscribe();
+
+  // Signal the correct reception of camera info
+  camera_info_received_ = true;
+  left_image_sub_->subscribe();
+  right_image_sub_->subscribe();
+}
+
+void StereoInterface::stereo_cb(
+  const Image::SharedPtr left_msg,
+  const Image::SharedPtr right_msg)
+{
+  rclcpp::Time left_stamp(left_msg->header.stamp);
+  rclcpp::Time right_stamp(right_msg->header.stamp);
+  if (left_stamp.nanoseconds() > last_stereo_timestamp_.nanoseconds()) {
+    static const VIO::CameraParams & left_cam_info =
+      vio_params_->camera_params_.at(0);
+    static const VIO::CameraParams & right_cam_info =
+      vio_params_->camera_params_.at(1);
+
+    const VIO::Timestamp & timestamp_left = left_stamp.nanoseconds();
+    const VIO::Timestamp & timestamp_right = right_stamp.nanoseconds();
+
+    // CHECK(left_frame_callback_)
+    // << "Did you forget to register the left frame callback?";
+    // CHECK(right_frame_callback_)
+    // << "Did you forget to register the right frame callback?";
+    //  TODO: Use RCLCPP_INFO inplace of CHECK?
+    // RCLCPP_INFO(this->get_logger(), "Did you forget to register the right frame callback?");
+
+    left_frame_callback_(VIO::make_unique<VIO::Frame>(
+        frame_count_, timestamp_left, left_cam_info, readRosImage(left_msg)));
```

Comment:
@ToniRV I saw a TODO comment in the KimeraVIO fronted when debugging https://github.com/MIT-SPARK/Kimera-VIO-ROS2/pull/5/commits/ec3586292209a953e671963e710ac84a6878f089 about the idea of removing the camera info argument from the frame callback.

https://github.com/ruffsl/Kimera-VIO/blob/e5e799c631b7ae3b8d3afac323996a381297348b/src/dataprovider/EurocDataProvider.cpp#L147-L149

 I just wanted to mention that keeping it might be nice for setups with changing extrinsic and intrinsic camera parameters.

E.g. When the camera has automatic focus or zoom, or when the camera to imu tranform is non-static, like with a robot's or drone articulated head with pan tilt control, but the IMU in fixed in place at the center of the chassis. These changes could still be picked up and synchronized with the data per-frame by consciously subscribing to camera_info and tf topics.